### PR TITLE
chore: trim deps and add dep checker step

### PR DESCRIPTION
**Pull request by mmastrac**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -52,6 +56,14 @@ jobs:
           cargo clippy --locked --release --all-features --all-targets -- -D clippy::all
           deno lint
 
+      - name: Lint deps
+        if: contains(matrix.os, 'ubuntu')
+        run: |
+          cargo install cargo-machete --locked
+          cargo machete
+          cargo install cargo-udeps --locked
+          cargo +nightly udeps
```
Comment (dsherret) [inline .github/workflows/ci.yml:65]: Consider putting this after cargo test so that people get faster feedback on failing tests.

**Review by dsherret — APPROVED** [review]
Nice!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mmastrac) [reply .github/workflows/ci.yml:68]: Good call, swapped.

Comment (mmastrac) [reply .github/workflows/ci.yml:68]: Actually, have to revert it because the tests cause a dep lint failure (because of compile tests :/)

**Pull request merged**